src/foam/core/XMLSupport.java

```diff
@@ -0,0 +1,158 @@
+package foam.core;
+
+
+import java.io.FileReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import com.sun.xml.internal.txw2.output.IndentingXMLStreamWriter;
+import foam.dao.XMLDAO;
+import foam.dao.ListSink;
+import java.io.FileWriter;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.*;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.ZoneId;
+
+public class XMLSupport {
+
+    // TODO: Planning to add support for other date formats in the future
+
+    // Parsing xml for FObjects and inserting into provided XMLDAO
+    public static void XMLToFObject (String fileName, XMLDAO xmlDAO) throws FileNotFoundException, XMLStreamException, IllegalAccessException {
```

Comment:
This shouldn't be tied to an XLDAO. I shouldn't have to implement this again if I need XML support for some other reason (networking, logging, GUI export, etc.).